### PR TITLE
fix: generate long desc and examples for root command

### PR DESCRIPTION
**Pull request by dvdksn**
We're already calling loadLongDescription recursively, no need to range over the subcommands, otherwise we're skipping the root command.

### CODE REVIEW

**Comment by dvdksn** [comment]
@crazy-max can you help me look at these test failures? I have tried generating docs with this version in both cli and buildx and I don't see any issues. I can't quite figure out what's going wrong in these tests either.

File: clidocstool.go [hunk 1]
```diff
@@ -88,33 +88,29 @@ func (c *Client) GenAllTree() error {
...
+	}
+	mdFile := strings.ReplaceAll(name, " ", "_") + ".md"
+	sourcePath := filepath.Join(c.source, mdFile)
+	content, err := os.ReadFile(sourcePath)
+	if os.IsNotExist(err) {
+		log.Printf("WARN: %s does not exist, skipping Markdown examples for %s docs\n", mdFile, generator)
+	}
+	if err != nil {
+		return err
+	}
```
Comment (crazy-max) [inline clidocstool.go:112]: ci error seems related to your changes: https://github.com/docker/cli-docs-tool/actions/runs/8661800382/job/23752482060?pr=53#step:5:23

You're returning the error where before we were continuing if the file didn't exist.

Comment (dvdksn) [reply clidocstool.go]: added a couple `return nil`, tests are happy.

**Review by crazy-max — APPROVED** [review]
LGTM

File: clidocstool_test.go [hunk 2]
```diff
@@ -227,6 +227,9 @@ format: "default|<id>[=<socket>|<key>[,<key>]]"`)
 func TestGenAllTree(t *testing.T) {
 	setup()
 	tmpdir := t.TempDir()
+	// tmpdir, err := os.MkdirTemp("", "clidocstool")
+	// require.NoError(t, err)
+	// t.Log(tmpdir)
```
Comment (thaJeztah) [inline clidocstool_test.go:232]: Looks like this is some debug code left behind

Comment (dvdksn) [reply clidocstool_test.go]: ah 🤦🏻 used this to update the fixtures. Removed it.

File: fixtures/docker.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# docker
```
Comment (thaJeztah) [inline fixtures/docker.md:1]: I see the man-page generated also includes `--help` as option, but don't see it in the Markdown variant here; is that expected, or is something missing?


It's possible that `--help` was an option that's added automatically (so not explicitly defined), so ... maybe related if that's the case?

File: fixtures/docker.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+command: docker
```
Comment (thaJeztah) [inline fixtures/docker.yaml:1]: Same for the YAML; do we expect this to include the `--help` flag (or at least custom-defined flags?)

File: fixtures/docker.md (hunk 3 above)
Comment (dvdksn) [reply fixtures/docker.md:1]: I wouldn't expect `--help` to show up as an option. It's an implicit flag in cobra, right? We don't define it as a flag anywhere so I'm not sure how the doc generation for it should work. I wonder how come it's being generated for the man page output, do you know @crazy-max ?

Comment (thaJeztah) [reply fixtures/docker.md:1]: Yeah, mostly noticed the difference between man-pages and YAML here. That said; the command-line will show `--help` as option, and given that these YAML-files are used to produce documentation, I somewhat would expect them to be complete;

```
      --help                             Print usage
```

I'm curious though of other flags _are_ included (see my comment below on the YAML file); is it because our root command didn't define any, or is something missing in the generator code to also include flags/options?

Comment (dvdksn) [reply fixtures/docker.md:1]: I'm thinking since the purpose of this PR is to enable longdesc generation for the `docker` base command, I don't think this discrepance between `man` and other formats needs to block. We don't have `--help` for the base command today, so there's no difference really.

**Comment by thaJeztah** [comment]
Sorry, I realise my question about `--help` somewhat distracted from the main issue; does this currently produce documentation for options/flags that are defined on the command? https://github.com/docker/cli-docs-tool/pull/53#discussion_r1576284060

I recall now I went looking for flags, and noticing the `--help` as example, but currently there's no flags in the produced output (but probably because none are defined). If we want to use this for the root command, we probably need to make sure that it includes flags 🙈

**Comment by dvdksn** [comment]
@thaJeztah Aaah, I see now yes. Good callout. I just pushed a commit that adds two flags:

- `--help` (hidden, persistent flag)
- `-H` (dummy flag)

I looked at the options implementation in docker/cli and turns out adding `--help` as a hidden flag also removes it from the fixtures (for subcommands as well, since it's a persistent flag). I think this change is appropriate since it mimics what we do in the core CLI.

https://github.com/docker/cli-docs-tool/pull/53/commits/0f4cffc674c9bfcef40067bf37d5c895459ae406#diff-9059958acfc03b30a2ff8fda8c5bb8403e63b17d7f32cdefe4611817db5cd191R57-R60

File: clidocstool_man_test.go [hunk 5]
```diff
@@ -43,7 +43,7 @@ func TestGenManTree(t *testing.T) {
 	c, err := New(Options{
 		Root:      dockerCmd,
 		SourceDir: tmpdir,
-		Plugin:    true,
+		Plugin:    false,
```
Comment (thaJeztah) [inline clidocstool_man_test.go:46]: Why were these changes needed? (Just curious as I couldn't immediately associate with the change in this PR)

Comment (dvdksn) [reply clidocstool_man_test.go:46]: Root commands for plugins are not generated; 

- https://github.com/docker/cli-docs-tool/blob/main/clidocstool_md.go#L50-L54
- https://github.com/docker/cli-docs-tool/blob/main/clidocstool_man.go#L61-L65
- https://github.com/docker/cli-docs-tool/blob/main/clidocstool_yaml.go#L110-L114

**Comment by dvdksn** [comment]
Let's merge this

**Review by crazy-max — APPROVED** [review]
LGTM

**Comment by dvdksn** [comment]
I've used this version to generate markdown here:

- docker/cli: https://github.com/docker/cli/pull/5010/commits/719f88ea6c09c4ab8d8c9974940e631dd55012a0
- docker/buildx: https://github.com/docker/buildx/pull/2565

I've also generated yaml (for docker/cli) here: https://github.com/docker/docs/pull/19794/commits/b9ca7336debd9e56e61a59d59ea51b74c8ac5760

**Comment by crazy-max** [comment]
Let's get this one in
